service/history/shard/context_impl.go

```diff
@@ -81,28 +79,14 @@ type (
 
 	ContextImpl struct {
 		// These fields are constant:
+		d                   ShardControllerDeps
```

Comment:
Just to confirm that we cannot embed this field? Maybe put a comment on why we cannot do that so other reader won't ask the same question.

Reply:
So, `ControllerImpl` can't embed `ShardControllerDeps`, since that would make it an `fx.In` itself, and `ControllerImpl` is a thing that's injected itself. It gives a runtime error:
```
Failed: cannot provide function "go.temporal.io/server/service/history/shard".ShardControllerProvider (/home/dnr/t/temporal/service/history/shard/fx.go:73): bad result 1: cannot provide parameter objects: *shard.ControllerImpl embeds a dig.In
```

However, `ContextImpl` is not injected, so it could embed `ShardControllerDeps`. I tried making that change: it does reduce the amount of `.d.`, although not the diff size since they changed names. It also means that those fields are all public fields of `ContextImpl`. But maybe that's okay since outside packages are supposed to use the `Context` interface (even though ContextImpl is public).

It seems a little better overall, I'll do that.